src/group_theory/submonoid/centralizer.lean

```diff
@@ -3,115 +3,25 @@ Copyright (c) 2021 Thomas Browning. All rights reserved.
-iff.rfl
-
-@[to_additive decidable_mem_add_centralizer]
-instance decidable_mem_centralizer [has_mul M] [decidable_eq M] [fintype M]
-  [decidable_pred (∈ S)] : decidable_pred (∈ centralizer S) :=
-λ _, decidable_of_iff' _ (mem_centralizer_iff)
-
-variables (S)
-
-@[simp, to_additive zero_mem_add_centralizer]
-lemma one_mem_centralizer [mul_one_class M] : (1 : M) ∈ centralizer S :=
-by simp [mem_centralizer_iff]
-
-@[simp]
-lemma zero_mem_centralizer [mul_zero_class M] : (0 : M) ∈ centralizer S :=
-by simp [mem_centralizer_iff]
-
-variables {S} {a b : M}
-
-@[simp, to_additive add_mem_add_centralizer]
-lemma mul_mem_centralizer [semigroup M] (ha : a ∈ centralizer S) (hb : b ∈ centralizer S) :
-  a * b ∈ centralizer S :=
-λ g hg, by rw [mul_assoc, ←hb g hg, ← mul_assoc, ha g hg, mul_assoc]
-
-@[simp, to_additive neg_mem_add_centralizer]
-lemma inv_mem_centralizer [group M] (ha : a ∈ centralizer S) : a⁻¹ ∈ centralizer S :=
-λ g hg, by rw [mul_inv_eq_iff_eq_mul, mul_assoc, eq_inv_mul_iff_mul_eq, ha g hg]
-
-@[simp]
-lemma add_mem_centralizer [distrib M] (ha : a ∈ centralizer S) (hb : b ∈ centralizer S) :
-  a + b ∈ centralizer S :=
-λ c hc, by rw [add_mul, mul_add, ha c hc, hb c hc]
-
-@[simp]
-lemma neg_mem_centralizer [has_mul M] [has_distrib_neg M] (ha : a ∈ centralizer S) :
-  -a ∈ centralizer S :=
-λ c hc, by rw [mul_neg, ha c hc, neg_mul]
-
-@[simp]
-lemma inv_mem_centralizer₀ [group_with_zero M] (ha : a ∈ centralizer S) : a⁻¹ ∈ centralizer S :=
-(eq_or_ne a 0).elim (λ h, by { rw [h, inv_zero], exact zero_mem_centralizer S })
-  (λ ha0 c hc, by rw [mul_inv_eq_iff_eq_mul₀ ha0, mul_assoc, eq_inv_mul_iff_mul_eq₀ ha0, ha c hc])
-
-@[simp, to_additive sub_mem_add_centralizer]
-lemma div_mem_centralizer [group M] (ha : a ∈ centralizer S) (hb : b ∈ centralizer S) :
-  a / b ∈ centralizer S :=
-begin
-  rw [div_eq_mul_inv],
-  exact mul_mem_centralizer ha (inv_mem_centralizer hb),
-end
-
-@[simp]
-lemma div_mem_centralizer₀ [group_with_zero M] (ha : a ∈ centralizer S) (hb : b ∈ centralizer S) :
-  a / b ∈ centralizer S :=
-begin
-  rw div_eq_mul_inv,
-  exact mul_mem_centralizer ha (inv_mem_centralizer₀ hb),
-end
-
-@[to_additive add_centralizer_subset]
-lemma centralizer_subset [has_mul M] (h : S ⊆ T) : centralizer T ⊆ centralizer S :=
-λ t ht s hs, ht s (h hs)
-
-variables (M)
-
-@[simp, to_additive add_centralizer_univ]
-lemma centralizer_univ [has_mul M] : centralizer univ = center M :=
-subset.antisymm (λ a ha b, ha b (set.mem_univ b)) (λ a ha b hb, ha b)
-
-variables {M} (S)
-
-@[simp, to_additive add_centralizer_eq_univ]
-lemma centralizer_eq_univ [comm_semigroup M] : centralizer S = univ :=
-subset.antisymm (subset_univ _) $ λ x hx y hy, mul_comm y x
-
-end set
-
 namespace submonoid
 section
-variables {M} [monoid M] (S)
+variables [monoid M] (S)
 
 /-- The centralizer of a subset of a monoid `M`. -/
 @[to_additive "The centralizer of a subset of an additive monoid."]
```

Comment:
Can you either:

* remove `subsemigroup.centralizer` from this PR, as its a little hard to review additions at the same time as moves
* add a `submonoid.center_to_subsemigroup` lemma to match [`subgroup.center_to_submonoid`](https://leanprover-community.github.io/mathlib_docs/group_theory/subgroup/basic.html#subgroup.center_to_submonoid)?

Reply:
I just did both. `to_additive` seemed to have trouble with some declarations because it apparently doesn't know to change `subsemigroup` to `add_subsemigroup`? Maybe there's a better way to handle it than what I did.